drivers/SmartThings/zigbee-switch/src/aqara/init.lua

```diff
@@ -70,9 +111,25 @@ local function is_aqara_products(opts, driver, device)
   return false
 end
 
-local function emit_power_consumption_report_event(device, value)
-  local raw_value = value.value -- 'Wh'
+local function round(num)
+  local mult = 10
+  return math.floor(num * mult + 0.5) / mult
+end
 
+local function energy_meter_power_consumption_report(device, raw_value)
+  -- report interval
+  local current_time = os.time()
+  local last_time = device:get_field(LAST_REPORT_TIME) or 0
+  local next_time = last_time + 60 * 15 -- 15 mins, the minimum interval allowed between reports
+  if current_time < next_time then
+    return
+  end
+  device:set_field(LAST_REPORT_TIME, current_time, { persist = true })
+
+  -- energy meter
+  device:emit_event(capabilities.energyMeter.energy({ value = raw_value, unit = "Wh" }))
```

Comment:
Shouldn't this be emitted regardless of the restrictions on `powerConsumptionReport`?

Reply:
The reported power consumption value not only used the energyMeter capability but also used the powerConsumptionReport capability. So I thought it would be good to display the event to the user at the same interval. Code changes required?

Reply:
`powerConsumptionReport` is more focused on historical consumption over periods of time, but for `energyMeter` you should just report the most up-to-date info you have

Reply:
I agree. However, I set it to 15 minute intervals in the configuration and timer condition because I need to consider for both current value and traffic.

```
local function present_value_handler(driver, device, value, zb_rx)
  # 195
  if current_time < next_time then
    return
  end
end

local function do_configure(self, device)
  # Line 242
  device:send(SimpleMetering.attributes.CurrentSummationDelivered:configure_reporting(device, 900, 3600, 1)) -- minimal interval : 15min
end
```